### PR TITLE
FEATURE: EGL-290 Support hidden/locked configuration for date filter

### PR DESCRIPTION
**Pull request by phong-nguyen-duy**
JIRA: EGL-290

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16300/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17811/ : SUCCESS in 11m 11s

File: libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx [hunk 1]
```diff
@@ -138,20 +144,38 @@ export const DateFilterCore: React.FC<IDateFilterCoreProps> = ({
...
-                                    dateFormat={verifiedDateFormat}
-                                    isTimeForAbsoluteRangeEnabled={isTimeForAbsoluteRangeEnabled}
-                                    weekStart={weekStart}
-                                />
-                            )}
+                            renderBody={({ closeDropdown }) => {
+                                return isConfigurationOpen ? (
+                                    <FilterConfigurationComponent
+                                        onSaveButtonClick={() => {
+                                            setIsConfigurationOpen(false);
```
Comment (vannguyengd) [inline libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx:151]: The inline function makes the component be rerenderred, please create a separate function for this prop!

File: libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx [hunk 2]
```diff
@@ -138,20 +144,38 @@ export const DateFilterCore: React.FC<IDateFilterCoreProps> = ({
...
+                                        }}
+                                        onCancelButtonClick={() => {
+                                            setIsConfigurationOpen(false);
+                                            onCancelClick();
```
Comment (vannguyengd) [inline libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx:155]: The inline function makes the component be rerenderred, please create a separate function for this prop!

File: libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx [hunk 3]
```diff
@@ -138,20 +144,38 @@ export const DateFilterCore: React.FC<IDateFilterCoreProps> = ({
...
+                                        filterOptions={filteredFilterOptions}
+                                        isMobile={isMobile}
+                                        closeDropdown={closeDropdown}
+                                        dateFilterButton={dateFilterButton()}
+                                        dateFormat={verifiedDateFormat}
+                                        isTimeForAbsoluteRangeEnabled={isTimeForAbsoluteRangeEnabled}
+                                        weekStart={weekStart}
+                                        isConfigurationEnabled={FilterConfigurationComponent !== undefined}
+                                        onConfigurationClick={() => {
+                                            setIsConfigurationOpen(true);
```
Comment (vannguyengd) [inline libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx:174]: The inline function makes the component be rerenderred, please create a separate function for this prop!

File: libs/sdk-ui-dashboard/src/presentation/filterBar/configuration/ConfigurationPanelHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// (C) 2023 GoodData Corporation
+import React from "react";
+
```
Comment (vannguyengd) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/configuration/ConfigurationPanelHeader.tsx:3]: We don't need an empty line to separate the import libraries.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/configuration/ConfigurationModeSelectItem.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+// (C) 2023 GoodData Corporation
+import React, { ChangeEvent } from "react";
+import cx from "classnames";
+
```
Comment (vannguyengd) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/configuration/ConfigurationModeSelectItem.tsx:4]: We don't need an empty line to separate the import libraries.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/configuration/ConfigurationModeSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
+// (C) 2023 GoodData Corporation
+import React, { ChangeEvent, useCallback } from "react";
+import { IntlShape } from "react-intl/src/types.js";
+import { ModeSelectItem } from "./ConfigurationModeSelectItem.js";
```
Comment (vannguyengd) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/configuration/ConfigurationModeSelect.tsx:4]: We need an empty line between lines 3 - 4 to separate the import libraries and import in the same project.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/dateFilter/DateFilterConfigurationBody.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
+// (C) 2023 GoodData Corporation
+import React, { useCallback, useState } from "react";
```
Comment (vannguyengd) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/dateFilter/DateFilterConfigurationBody.tsx:2]: It is better if you can group the import libraries and import in the same project into separate groups.

File: libs/sdk-ui-filters/src/DateFilter/index.ts [hunk 8]
```diff
@@ -72,6 +72,8 @@ export {
     isUiRelativeDateFilterForm,
 } from "./interfaces/index.js";
 
+export { IFilterConfigurationProps } from "./DateFilterBody/types.js";
+
```
Comment (vannguyengd) [inline libs/sdk-ui-filters/src/DateFilter/index.ts:76]: I think we don't need the empty lines to separate the imports if they are on the same level.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16310/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17821/ : FAILURE in 8m 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16312/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17824/ : FAILURE in 11m 54s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16318/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17833/ : SUCCESS in 10m 19s

**Review by vannguyengd — APPROVED** [review]
LGTM!

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4245

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3358/ : SUCCESS in 10m 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17839/ : SUCCESS in 10m 35s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/966/ : SUCCESS in 12m 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4521/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1238/ : FAILURE in 1m 35s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/870/ : SUCCESS in 10m 43s

**Pull request merged**